Looks like rdma-core lib dependency missing on build machine
`In file included from /home/runner/work/ntirpc/ntirpc/src/clnt_vc.c:76:
/home/runner/work/ntirpc/ntirpc/src/rpc_rdma.h:43:10: fatal error: rdma/rdma_cma.h: No such file or directory
   43 | #include <rdma/rdma_cma.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.`

--- a/src/clnt_rdma.c
+++ b/src/clnt_rdma.c
@@ -159,11 +159,13 @@ clnt_rdma_ncreatef(const SVCXPRT *xprt,		/* init but NOT connect()ed */
 		if (rpc_rdma_connect(rdma_xprt)) {
 			__warnx(TIRPC_DEBUG_FLAG_ERROR,
 				"%s: rdma connect failed", __func__);
+			cl->cl_error.re_status = RPC_UNKNOWNADDR;
 			return (cl);
 		}
 		if (rpc_rdma_connect_finalize(rdma_xprt)) {
 			__warnx(TIRPC_DEBUG_FLAG_ERROR,
 				"%s: rdma connect finalize failed", __func__);
+			cl->cl_error.re_status = RPC_UNKNOWNADDR;
 			return (cl);
 		}
 
@@ -178,6 +180,7 @@ clnt_rdma_ncreatef(const SVCXPRT *xprt,		/* init but NOT connect()ed */
 		if (xdr_rdma_create(rdma_xprt)) {
 			__warnx(TIRPC_DEBUG_FLAG_ERROR,
 				"%s: buffer allocation failed", __func__);
+			cl->cl_error.re_status = RPC_SYSTEMERROR;
 			return (cl);
 		}
 	}

--- a/src/clnt_vc.c
+++ b/src/clnt_vc.c
@@ -73,7 +73,9 @@
 #include "svc_ioq.h"
 #include "clnt_internal.h"
 #include "svc_internal.h"
+#ifdef USE_RPC_RDMA
 #include "rpc_rdma.h"
+#endif
 
 static enum xprt_stat clnt_vc_process(struct svc_req *req);
 static struct clnt_ops *clnt_vc_ops(void);
@@ -125,7 +127,7 @@ clnt_vc_data_zalloc(void)
  *      server tranpsorts sharing an underlying bytestream (Matt).
  */
 
-
+#ifdef USE_RPC_RDMA
 /* Create new RDMA client with specified connection parameters */
 CLIENT *
 clnt_rdma_create(int fd, char *host, int port, int recv_sz, int send_sz,
@@ -166,6 +168,7 @@ clnt_rdma_create(int fd, char *host, int port, int recv_sz, int send_sz,
 	/* create rdma_xprt using xprt */
 	return clnt_rdma_ncreatef(xprt, prog, vers, flags, true);
 }
+#endif
 
 /*
  * Create a client handle for a connection.

--- a/src/rpc_rdma.c
+++ b/src/rpc_rdma.c
@@ -2454,7 +2454,7 @@ rpc_rdma_bind_client(RDMAXPRT *rdma_xprt)
 
 		rc = rdma_resolve_addr(rdma_xprt->cm_id, res->ai_src_addr,
 					res->ai_dst_addr,
-					__svc_params->idle_timeout);
+					5000);
 		if (rc) {
 			rc = errno;
 			__warnx(TIRPC_DEBUG_FLAG_ERROR,
@@ -2474,7 +2474,7 @@ rpc_rdma_bind_client(RDMAXPRT *rdma_xprt)
 		}
 
 		rc = rdma_resolve_route(rdma_xprt->cm_id,
-					__svc_params->idle_timeout);
+					5000);
 		if (rc) {
 			rdma_xprt->state = RDMAXS_ERROR;
 			__warnx(TIRPC_DEBUG_FLAG_ERROR,

--- a/tests/rpcping.c
+++ b/tests/rpcping.c
@@ -375,12 +375,17 @@ int main(int argc, char *argv[])
 				perror("get_conn_fd failed");
 				exit(3);
 			}
-			clnt = clnt_rdma_create(fd, "10.53.87.150", 20049, recv_sz,
+#ifdef USE_RPC_RDMA
+			clnt = clnt_rdma_create(fd, host, 20049, recv_sz,
 			    send_sz, page_sz, prog, vers, CLNT_CREATE_FLAG_CLOSE);
 			if (CLNT_FAILURE(clnt)) {
 				rpc_perror(&clnt->cl_error, "clnt_rdma_create failed");
 				exit(4);
 			}
+#else
+			perror("rdma not enabled");
+			exit(4);
+#endif
 		}
 
 		s = &states[i];

